### ISSUE TITLE
Forced upgrade for history feature

### ISSUE THREAD

**Issue by PandaScience** [sharelatex/sharelatex#275]
Hi @ all,

I just tried to install sharelatex on our institute's webserver, at this point: 
**thx a lot to the developers for sharing and making it open-source!!**

I mostly followed the installation instructions from the manual and finally got it to work...kind of. When trying to open a test project, it tells
>You need to upgrade your account to use the History feature. 
>Start Free Trial! -- Cancel`
>Please refresh this page after starting your free trial.

and

>Please ask the project owner to upgrade to use the History feature.
>Cancel

and after clicking any of the buttons, it will reload and show the same stuff again. 

some specs:
- running on a debian jessie (3.16.0-4-amd64) in KVM / libvirt
- mongodb v2.6.8 (debian package)
- nodejs v0.10.36 (official repo)
- redis-server v2.8.17 (debian package)
- proxy is nginx v1.6.2 (debian package)

I'm kind of new to all these software...so no idea where to start looking for possible errors. Any ideas?

Thanks for your help,
Rene

**Comment by henryoswald** [comment]
Hi Rene,

it sounds like either the javascript is not loading or you have not set the defaultFeatures in settings to have versioning to true.

**Comment by PandaScience** [comment]
Hi again,

ok I grep-ed "versioning" and found the defaultFeatures entry in 
> %sharelatex-root%/web/config/settings.defaults.coffee:143 --> versioning: true

How could I check the javascript stuff? 

Btw. 
> grunt check --force 

also tells 2 warnings, which I've ignored at first. maybe has to do sth. with that?

>Running "check:s3" task
>Warning: Cannot read property 'backend' of undefined Used --force, continuing.
>
>Running "check:fs" task
>Warning: Cannot read property 'backend' of undefined Used --force, continuing.

**Comment by Jordi-** [comment]
Hi.

Same issue here. In my case I'm running the vagrant ubuntu 12.04 machine. Everything installed correctly, but when you create a new project it complains about the History feature. As PandaScience, the user I created has versioning set to true (I check this out in sharelatex mongodb).

**Comment by Jordi-** [comment]
Hi again.

Just to report that I have installed sharelatex on ubuntu 12.04 this time using the recommended docker image. The installation works OK, but when trying to create a new project it fails with the same error.

You can see the reply in the attached image. The history feature makes little sense since this is a new project. Also it seems like a everything is undefined (project name, messages, etc).

![sharelatex-2015-03-02 13 39 37](https://cloud.githubusercontent.com/assets/8670324/6440899/43f030f8-c0e2-11e4-91cb-48b46a524ece.png)

**Comment by jpallen** [comment]
Have you followed the Docker instructions and set the SHARELATEX_SITE_URL environment variable to point to the same address as you have in your browser address bar when accessing the site?

**Comment by henryoswald** [comment]
It looks like your client side javascript is not loading up. One
possibility is you are blocking scripts.

On Mon, Mar 2, 2015 at 12:49 PM, James Allen <notifications@github.com>
wrote:

> Have you followed the Docker instructions and set the SHARELATEX_SITE_URL
> environment variable to point to the same address as you have in your
> browser address bar when accessing the site?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sharelatex/sharelatex/issues/275#issuecomment-76706144>
> .
>



-- 
Henry Oswald
shareLaTeX.com creator

**Comment by PandaScience** [comment]
I already tried different browsers, with and without "no-script" plugins, also one on a fresh debian installation, all failed. 

Btw, here I found a new installation tutorial. Haven't had a look yet, but maybe they found sth. useful to contribute. Unfortunately, it's written in german but you can just look at the code lines.

http://www.freiesmagazin.de/ftp/2015/freiesMagazin-2015-03.pdf

and again thx for your help,...really appreciating it :)

**Comment by Jordi-** [comment]
Thanks for you answers.

Setting SHARELATEX_SITE_URL partially fix the problem. It was my fault because the instructions are very clear about this.

Now creating a new project works, but when  it opens it gives 'Error opening document', and informs that 'something went wrong opening this document.' In the PDF preview there is also an error saying that something went wrong too. Any text file can be open or deleted, although they can be renamed. On the other hand the file 'universe.jpg' can be opened and renamed, but not deleted.

New files can be uploaded, but once on the server they exhibit the same behavior as explained above, that is, text files do not work, images do, neither can be deleted. 

All these errors are for the docker installation.

On the other hand, when installing without docker following the instructions at https://github.com/sharelatex/sharelatex/wiki/Production-Installation-Instructions everything seems to work OK using nginx as a reverse proxy, provided you set correctly the following directives in settings.coffe:

  - siteUrl : 'http://localhost:5000' # <= change to you listening port, as in docker
  - websocketsUrl: 'http://localhost:5000' # <= should be the same
  - behindProxy: true

Regarding the client side javacript not loading, I can confirm that adblock does not interfere with sharelatex, but ublock for chrome does. However, it does not raise the History feature error, it just does not show the login dialog, so you cannot even enter the site.

**Comment by Jordi-** [comment]
I forgot to ask this: Is there any way to see the logs on the docker installation in order to try to figure out what's going on?

**Comment by jpallen** [comment]
You can get into the docker container with:

```
docker exec -i -t sharelatex /bin/bash
```

The logs are in /var/log/sharelatex.

**Comment by Jordi-** [comment]
OK, I found a couple of ways of debugging the docker session. Besides the one appointed by James (thanks :+1:), it is possible to mount /var/log/sharelatex in a volume, or even better use nsenter.

Anyway, the problem with docker was again my fault :-(. It turned out that I was using an older version of redis-server, the one that comes with ubuntu 12.04. I updated to 2.8 and now everything works fine. mongo was also an older version than the one recommended, but it works anyway. In any case I have updated it too to 2.6 to make sure everything goes OK.

As a curiosity, ad blockers do not seem to interfere with the docker version.

This makes all my problems (caused by me :-s) solved. Thanks for your support :-).
